dbt_platform_helper/utils/aws.py

```diff
@@ -351,6 +356,61 @@ def get_postgres_connection_data_updated_with_master_secret(session, parameter_n
     return parameter_data
 
 
+def get_supported_redis_versions(elasticache_client=None):
+
+    if cache_refresh_required("redis"):
+
+        supported_versions = []
+
+        if not elasticache_client:
```

Comment:
I am still unhappy about this existing becuase we need to pass the client in for the tests.

I would prefer to remove the `elasticache_client=None` method argument and condition and patch `elasticache_client.describe_cache_engine_versions` for this pull request, then deal with it properly via depencency injection during the work where we are extracting the AWS provider.